code/modules/mob/living/simple_animal/abnormality/he/scaredy_cat.dm

```diff
@@ -0,0 +1,165 @@
+		if(!can_see(src, friend, vision_range))
+			GoToFriend()
+
+/mob/living/simple_animal/hostile/abnormality/scaredy_cat/proc/ProtectFriend(mob/living/simple_animal/hostile/abnormality/abno)
+	wait_for_friend = FALSE
+	Courage(TRUE)
+	friend = abno //You are friend =D
+	faction = friend.faction
+	GoToFriend()
+
+/mob/living/simple_animal/hostile/abnormality/scaredy_cat/death()
+	density = FALSE
+	if(friend)
+		addtimer(CALLBACK(src, .proc/Regenerate), 20 SECONDS)
+	else
+		animate(src, alpha = 0, time = 10 SECONDS)
+		QDEL_IN(src, 10 SECONDS)
+	..()
+
+/mob/living/simple_animal/hostile/abnormality/scaredy_cat/Destroy()
+	UnregisterSignal(SSdcs, COMSIG_GLOB_MOB_DEATH)
+	UnregisterSignal(SSdcs, COMSIG_GLOB_ABNORMALITY_BREACH)
+	return ..()
+
+//if he still has a friend, revives
+/mob/living/simple_animal/hostile/abnormality/scaredy_cat/proc/Regenerate()
+	if(friend)
+		density = TRUE
+		revive(full_heal = TRUE, admin_revive = TRUE)
+		GoToFriend()
+		return
+	animate(src, alpha = 0, time = 3 SECONDS)
+	QDEL_IN(src, 3 SECONDS)
+
+///teleports to his friend if he can't see them
+/mob/living/simple_animal/hostile/abnormality/scaredy_cat/proc/GoToFriend()
+	if(!friend)
+		return
+	var/turf/origin = get_turf(friend)
+	var/list/all_turfs = RANGE_TURFS(2, origin)
+	for(var/turf/T in all_turfs)
+		if(!T.density && T != origin && can_see(origin, T, friend.vision_range))
+			playsound(src, 'sound/abnormalities/scaredycat/cateleport.ogg', 50, FALSE, 4)
+			forceMove(T)
+			for(var/mob/living/carbon/human/enemy in view(vision_range, src))
+				GiveTarget(enemy) //the moment he teleports he's already on the offensive
+				return
+
+///If scaredy cat becomes a big boy or a baby boy
+/mob/living/simple_animal/hostile/abnormality/scaredy_cat/proc/Courage(courage)
+	if(courage)
+		melee_damage_lower = 20
+		melee_damage_upper = 25
+		damage_coeff = list(BRUTE = 1, RED_DAMAGE = 0.5, WHITE_DAMAGE = 2, BLACK_DAMAGE = 1.5, PALE_DAMAGE = 2)
+	else
+		faction = list("neutral")
+		melee_damage_lower = 1
+		melee_damage_upper = 1 //it shouldn't attack in that form in the first place but...
+		damage_coeff = list(BRUTE = 1, RED_DAMAGE = 4, WHITE_DAMAGE = 4, BLACK_DAMAGE = 4, PALE_DAMAGE = 4)
+		playsound(src, 'sound/abnormalities/scaredycat/catchange.ogg', 75, FALSE, 4)
+
+/mob/living/simple_animal/hostile/abnormality/scaredy_cat/proc/OnMobDeath(datum/source, mob/living/died, gibbed)
+	SIGNAL_HANDLER
+	if(!friend || (status_flags & GODMODE))
+		return FALSE
+	if(died == friend)
+		friend = null
+		Courage(FALSE)
+		if(stat != DEAD)
+			wait_for_friend = TRUE //kill him fast before he finds another friend
+		return TRUE
+	return FALSE
+
+/mob/living/simple_animal/hostile/abnormality/scaredy_cat/proc/OnAbnoBreach(datum/source, mob/living/simple_animal/hostile/abnormality/abno)
+	SIGNAL_HANDLER
+	if(status_flags & GODMODE)
+		if(abno.name == "Scarecrow searching for wisdom" || abno.name == "Warm-Hearted Woodsman") //this will get crowded when more oz abno get added
```

Comment:
To make this more efficient, make a list of Abno names that it should look for and then check for "abno.name in prefered_abno_list"